x-pack/plugins/alerting/common/routes/rule/apis/create/schemas/v1.ts

```diff
@@ -23,7 +24,14 @@ export const actionAlertsFilterSchema = schema.object({
         schema.object({
           query: schema.maybe(schema.recordOf(schema.string(), schema.any())),
           meta: schema.recordOf(schema.string(), schema.any()),
-          state$: schema.maybe(schema.object({ store: schema.string() })),
+          $state: schema.maybe(
+            schema.object({
+              store: schema.oneOf([
+                schema.literal(filterStateStore.APP_STATE),
+                schema.literal(filterStateStore.GLOBAL_STATE),
```

Comment:
we should not allow the Global_State, we do not want this filter to become global to any search bar.